src/System.Drawing.Primitives/src/System/Drawing/Rectangle.cs

```diff
@@ -0,0 +1,474 @@
+        ///       Gets the x-coordinate of the upper-left corner of the
+        ///       rectangular region defined by this <see cref='System.Drawing.Rectangle'/> .
+        ///    </para>
+        /// </summary>
+        public int Left
+        {
+            get
+            {
+                return X;
+            }
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Gets the y-coordinate of the upper-left corner of the
+        ///       rectangular region defined by this <see cref='System.Drawing.Rectangle'/>.
+        ///    </para>
+        /// </summary>
+        public int Top
+        {
+            get
+            {
+                return Y;
+            }
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Gets the x-coordinate of the lower-right corner of the
+        ///       rectangular region defined by this <see cref='System.Drawing.Rectangle'/>.
+        ///    </para>
+        /// </summary>
+        public int Right
+        {
+            get
+            {
+                return X + Width;
+            }
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Gets the y-coordinate of the lower-right corner of the
+        ///       rectangular region defined by this <see cref='System.Drawing.Rectangle'/>.
+        ///    </para>
+        /// </summary>
+        public int Bottom
+        {
+            get
+            {
+                return Y + Height;
+            }
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Tests whether this <see cref='System.Drawing.Rectangle'/> has a <see cref='System.Drawing.Rectangle.Width'/>
+        ///       or a <see cref='System.Drawing.Rectangle.Height'/> of 0.
+        ///    </para>
+        /// </summary>
+        public bool IsEmpty
+        {
+            get
+            {
+                return _height == 0 && _width == 0 && _x == 0 && _y == 0;
+            }
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Tests whether <paramref name="obj"/> is a <see cref='System.Drawing.Rectangle'/> with
+        ///       the same location and size of this Rectangle.
+        ///    </para>
+        /// </summary>
+        public override bool Equals(object obj)
+        {
+            if (!(obj is Rectangle))
+                return false;
+
+            Rectangle comp = (Rectangle)obj;
+
+            return (comp.X == X) && (comp.Y == Y) && (comp.Width == Width) && (comp.Height == Height);
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Tests whether two <see cref='System.Drawing.Rectangle'/>
+        ///       objects have equal location and size.
+        ///    </para>
+        /// </summary>
+        public static bool operator ==(Rectangle left, Rectangle right)
+        {
+            return (left.X == right.X && left.Y == right.Y &&
+                left.Width == right.Width && left.Height == right.Height);
```

Comment:
There are a good few cases like this where multiple branches are done on what are likely relatively fast comparisons that could hence perhaps be faster as `left.X == right.X & left.Y == right.Y & left.Width == right.Width & left.Height == right.Height`, etc. Though if such a change was considered worthwhile it's probably best left to a separate change with some profiling to see if they actually do gain anything.